scripts/zkai_attention_kv_stwo_fine_grained_component_schema_gate.py

```
@@ -0,0 +1,829 @@
+    try:
+        return json.loads(completed.stdout)
+    except json.JSONDecodeError as err:
+        raise StwoFineGrainedComponentSchemaGateError(f"Stwo proof component schema CLI did not emit JSON: {err}") from err
+
+
+def artifact_proof_sha256(path: str) -> str:
+    resolved = (ROOT / path).resolve()
+    try:
+        envelope = json.loads(resolved.read_text(encoding="utf-8"))
+    except OSError as err:
+        raise StwoFineGrainedComponentSchemaGateError(f"failed to read artifact envelope {path}: {err}") from err
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)